core-bundle/src/Resources/contao/library/Contao/Controller.php

```diff
@@ -1512,6 +1518,226 @@ protected function eliminateNestedPages($arrPages, $strTable=null, $blnSorting=f
+				}
+				else
+				{
+					$maxWidth -= $horizontalMargin;
+				}
+			}
+
+			// Normalize size
+			if ($size instanceof PictureConfiguration)
+			{
+				return array($size, $margin);
+			}
+
+			if (is_numeric($size))
+			{
+				$size = array(0, 0, (int) $size);
+			}
+			else
+			{
+				$size = (\is_array($size) ? $size : array()) + array(0, 0, 'crop');
+				$size[0] = (int) $size[0];
+				$size[1] = (int) $size[1];
+			}
+
+			// Adjust image size configuration if it exceeds the max width
+			if ($size[0] > 0 && $size[1] > 0)
+			{
+				list($width, $height) = $size;
+			}
+			else
+			{
+				$container = System::getContainer();
+
+				/** @var BoxInterface $originalSize */
+				$originalSize = $container
+					->get('contao.image.image_factory')
+					->create($container->getParameter('kernel.project_dir') . '/' . $rowData['singleSRC'])
+					->getDimensions()
+					->getSize();
+
+				$width = $originalSize->getWidth();
+				$height = $originalSize->getHeight();
+			}
+
+			if ($width <= $maxWidth)
+			{
+				return array($size, $margin);
+			}
+
+			$size[0] = $maxWidth;
+			$size[1] = floor($maxWidth * ($height / $width));
+
+			return array($size, $margin);
+		};
+
+		/** @var FigureBuilder $figureBuilder */
+		$figureBuilder = System::getContainer()->get(Studio::class)->createFigureBuilder();
+
+		// Set image resource
+		try
+		{
+			if (null !== $filesModel)
+			{
+				// Make sure model points to the same resource (BC)
+				$filesModel = clone $filesModel;
+				$filesModel->path = $rowData['singleSRC'];
+
+				// Use source + meta data from files model (if not overwritten)
+				$figureBuilder
+					->fromFilesModel($filesModel)
+					->setMetaData($createMetaDataOverwriteFromRowData(true));
+
+				$includeFullMetaData = true;
+			}
+			else
+			{
+				// Always ignore file meta data when building from path (BC)
+				$figureBuilder
+					->fromPath($rowData['singleSRC'], false)
+					->setMetaData($createMetaDataOverwriteFromRowData(false));
+
+				$includeFullMetaData = false;
+			}
+		}
+		catch (InvalidResourceException $e)
+		{
+			System::getContainer()
+				->get('monolog.logger.contao')
+				->log(
+					LogLevel::ERROR,
+					sprintf('Image "%s" could not be processed: %s', $rowData['singleSRC'], $e->getMessage()),
+					array('contao' => new \Contao\CoreBundle\Monolog\ContaoContext(__METHOD__, TL_ERROR))
+				);
+
+			// Fall back to apply a sparse data set instead of failing (BC)
+			foreach ($createFallBackTemplateData() as $key => $value)
+			{
+				$template->$key = $value;
+			}
+
+			return;
+		}
+
+		// Set size and light box configuration
+		list($size, $margin) = $getSizeAndMargin();
+
+		$lightBoxSize = StringUtil::deserialize($rowData['lightboxSize'] ?? null) ?: null;
+
+		$figureBuilder
+			->setSize($size)
+			->setLightBoxGroupIdentifier($lightBoxGroupIdentifier)
+			->setLightBoxSize($lightBoxSize)
+			->enableLightBox('1' === ($rowData['fullsize'] ?? null));
+
+		// todo: $rowData['lightboxSize'] + test
```

Comment:
`$rowData['lightboxSize']` is already handled on line 1717, forgot to remove this comment?

Reply:
Ah, yes that's a leftover. Fixed in 3fb26ab92877c899d6a21dfe8c4a64c638d56bfa.